src/systems/systems/src/pneumatic/mod.rs

```diff
@@ -128,32 +218,53 @@ impl PneumaticPipe {
     }
 
     fn change_volume(&mut self, new_volume: Volume) {
+        let new_pressure = self.calculate_pressure_for_new_volume(new_volume);
+        self.update_temperature_for_target_pressure(new_pressure);
+        self.pressure = new_pressure;
         self.volume = new_volume;
     }
```

Comment:
Is there a situation in which we wish to change the volume of a `PneumaticPipe` at runtime, now that we use mass as a measure for fluid quantity? We have a `VariableVolumeContainer` which should serve this exact purpose.

Reply:
I think the only location where it's used is in `VariableVolumeContainer`. I didn't want to change that. But I think it should be possible.

Reply:
Yeah, in general, we don't want the volume (i.e. the physical volume, like the available space) of a container to change. The exception is the `VariableVolumeContainer `, which is used for hydraulic reservoirs, where the amount of space that is available to the bleed air, depends on the amount of hydraulic fluid present in the reservoir.